File: src/main/java/com/architecture/microservices/domain/repository/FileItemRepository.java
package com.architecture.microservices.domain.repository;

import com.architecture.microservices.domain.model.FileItemEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface FileItemRepository extends JpaRepository<FileItemEntity, String> {
    List<FileItemEntity> findByParentId(String parentId);
    List<FileItemEntity> findByParentIdIsNull();
}

File: src/main/kotlin/com/architecture/microservices/domain/repository/FileItemRepository.kt
package com.architecture.microservices.domain.repository

import com.architecture.microservices.domain.model.FileItemEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface FileItemRepository : JpaRepository<FileItemEntity, String> {
    fun findByParentId(parentId: String?): List<FileItemEntity>
    fun findByParentIdIsNull(): List<FileItemEntity>
}
